orangecontrib/network/community.py

```diff
@@ -181,7 +181,14 @@ def next_label(neighbors):
             if stop:
                 break
 
+    labels = remap_labels(labels)
+
     if results2items:
-        add_results_to_items(G, lblhistory)
+        add_results_to_items(G, labels)
 
     return labels
+
+def remap_labels(labels):
+    unique_labels = sorted(set(labels.values()))
+    dict_labels = dict(zip(unique_labels, range(len(unique_labels))))
+    return dict(zip(labels.keys(), [dict_labels[x] for x in labels.values()]))
```

Comment:
Here I'd definitely use dict comprehension: `return {key: dict_labels[val] for key, val in labels.items()}`.

Also, avoid using `keys()` method. Iteration over a dict already gives the keys.

The second argument could be a generator instead of a list.